spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -104,8 +101,16 @@ else if (tokens.length != 1) {
 					channel = new DirectChannel();
 					this.beanFactory.registerSingleton(beanName, channel);
 					channel = (MessageChannel) this.beanFactory.initializeBean(channel, beanName);
-					Binder<MessageChannel> binder = binderFactory.getBinder(transport);
-					this.dynamicDestinationsBindable.addOutputBinding(beanName, binder.bindProducer(destinationName, channel, producerProperties));
+					@SuppressWarnings("unchecked")
+					Binder<MessageChannel, ?, ProducerProperties> binder =
+							(Binder<MessageChannel, ?, ProducerProperties>) binderFactory.getBinder(binderName);
+					// TODO: need the props to return some defaults if not found
+					Class<? extends ProducerProperties> producerPropertiesClass =
+							ChannelBindingService.resolveProducerPropertiesType(binder);
+					ProducerProperties producerProperties =
+							this.channelBindingServiceProperties.getProducerProperties(destinationName, producerPropertiesClass);
```

Comment:
the first param of `getProducerProperties(..)` is *channelName*, but here the "destinationName" is passed... is that intentional?